### PR TITLE
Add waydroid

### LINKED ISSUE #580: adds waydroid setup
## Is your feature request related to a problem? Please describe.
every time i try to setup waydroid manually it always fail

## Describe the solution you'd like
i want an option to setup waydroid

## Describe alternatives you've considered
non

## Additional context
https://waydro.id/

### PR DESCRIPTION
**Pull request by nnyyxxxx**
## Type of Change
- [x] New feature

## Description
adds waydroid, uses instructions from https://docs.waydro.id/usage/install-on-desktops for installation

## Testing
tested

## Issues / other PRs related

- Resolves #580 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/waydroid-setup.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: $PACKAGER${RC}"
+                exit 1
+                ;;
+        esac
+    else
```
Comment (adamperkowski) [inline core/tabs/applications-setup/waydroid-setup.sh:24]:
```suggestion
        esac
    printf "%b\n" "${GREEN}Waydroid installation complete.${RC}"
    else
```

**Review by adamperkowski — APPROVED** [review]

File: core/tabs/applications-setup/waydroid-setup.sh (hunk 1 above)
Comment (nnyyxxxx) [reply core/tabs/applications-setup/waydroid-setup.sh:24]: that would break the consistency with the other application scripts

**Comment by adamperkowski** [comment]
We should consider this if system is using dkms
https://www.reddit.com/r/linux4noobs/comments/rdpsz4/driver_problem_while_attempting_to_install/

File: core/tabs/applications-setup/waydroid-setup.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+installWaydroid() {
+    if ! command_exists waydroid; then
+    printf "%b\n" "${YELLOW}Installing Waydroid...${RC}"
+        case "$PACKAGER" in
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm waydroid # dont run any aur helper as root
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/waydroid-setup.sh:10]:
```suggestion
                "$AUR_HELPER" -S --needed --noconfirm waydroid
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/waydroid-setup.sh (hunk 2 above)
Comment (jeevithakannan2) [reply core/tabs/applications-setup/waydroid-setup.sh:10]: I don't this that comment is necessary there

Comment (nnyyxxxx) [reply core/tabs/applications-setup/waydroid-setup.sh:10]: i usually include context for the maintainer, but i see your point

**Comment by jeevithakannan2** [comment]
Waydroid only works on wayland so add pre-condition in toml

**Comment by jeevithakannan2** [comment]
Also it won't work with nvidia GPU it would be better if we notified the user about that and make a work around to use software rendering if nvidia GPU is used

**Comment by nnyyxxxx** [comment]
> Waydroid only works on wayland so add pre-condition in toml

sure:)

**Comment by nnyyxxxx** [comment]
needs 2 issues to be worked out before merge

File: core/tabs/applications-setup/waydroid-setup.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm waydroid
+                ;;
+            apt-get|nala)
+                curl https://repo.waydro.id | "$ESCALATION_TOOL" sh
+                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
+                ;;
```
Comment (adamperkowski) [inline core/tabs/applications-setup/waydroid-setup.sh:18]:
```suggestion
            pacman)
                "$AUR_HELPER" -S --needed --noconfirm waydroid
                if command_exists dkms; then
                    "$AUR_HELPER" -S --needed --noconfirm binder_linux-dkms
                    "$ESCALATION_TOOL" modprobe binder-linux device=binder,hwbinder,vndbinder
                fi
                ;;
            apt-get|nala)
                curl https://repo.waydro.id | "$ESCALATION_TOOL" sh
                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
                if command_exists dkms; then
                    "$ESCALATION_TOOL" "$PACKAGER" install -y git
                    git clone https://github.com/choff/anbox-modules.git && cd anbox-modules
                    "$ESCALATION_TOOL" cp anbox.conf /etc/modules-load.d/
                    "$ESCALATION_TOOL" cp 99-anbox.rules /lib/udev/rules.d/
                    "$ESCALATION_TOOL" cp -rT ashmem /usr/src/anbox-ashmem-1
                    "$ESCALATION_TOOL" cp -rT binder /usr/src/anbox-binder-1
                    "$ESCALATION_TOOL" dkms install anbox-ashmem/1
                    "$ESCALATION_TOOL" dkms install anbox-binder/1
                fi
                ;;
            dnf)
                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
                if command_exists dkms; then
                    "$ESCALATION_TOOL" "$PACKAGER" install -y git
                    git clone https://github.com/choff/anbox-modules.git && cd anbox-modules
                    "$ESCALATION_TOOL" cp anbox.conf /etc/modules-load.d/
                    "$ESCALATION_TOOL" cp 99-anbox.rules /lib/udev/rules.d/
                    "$ESCALATION_TOOL" cp -rT ashmem /usr/src/anbox-ashmem-1
                    "$ESCALATION_TOOL" cp -rT binder /usr/src/anbox-binder-1
                    "$ESCALATION_TOOL" dkms install anbox-ashmem/1
                    "$ESCALATION_TOOL" dkms install anbox-binder/1
                fi
                ;;
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]
@nnyyxxxx Something like this. Feel free to restructure my changes.

File: core/tabs/applications-setup/tab_data.toml [hunk 4]
```diff
@@ -74,6 +74,16 @@ name = "Rofi"
...
 
+[[data]]
+name = "Waydroid"
+description = "Waydroid is an emulator that allows you to run Android apps and games on Linux."
+script = "waydroid-setup.sh"
+
+[[data.preconditions]]
+matches = true
+data = "command_exists"
+values = [ "wayland-info", "weston" ]
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml:85]: Do something like this but with `WAYLAND_DISPLAY` instead:
https://github.com/ChrisTitusTech/linutil/blob/995a51a66548b1f913bafddbe5241d2ebdf778ca/core/tabs/utils/tab_data.toml#L7-L10

Comment (nnyyxxxx) [reply core/tabs/applications-setup/tab_data.toml:85]: tahnks

**Review by adamperkowski — APPROVED** [review]

**Comment by nnyyxxxx** [comment]
ready to be merged

File: core/tabs/applications-setup/waydroid-setup.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                    "$ESCALATION_TOOL" cp 99-anbox.rules /lib/udev/rules.d/
+                    "$ESCALATION_TOOL" cp -rT ashmem /usr/src/anbox-ashmem-1
+                    "$ESCALATION_TOOL" cp -rT binder /usr/src/anbox-binder-1
+                    "$ESCALATION_TOOL" dkms install anbox-ashmem/1
+                    "$ESCALATION_TOOL" dkms install anbox-binder/1
+                fi
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y waydroid
+                if command_exists dkms; then
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/waydroid-setup.sh:34]: these if statements are a bit messy, can be cleaned up in the future.

**Review by adamperkowski — APPROVED** [review]

**Comment by jeevithakannan2** [comment]
@nnyyxxxx waydroid won't work with nvidia GPU it would be better if we notified the user about that and make a work around to use software rendering if nvidia GPU is used

**Comment by nnyyxxxx** [comment]
> @nnyyxxxx waydroid won't work with nvidia GPU it would be better if we notified the user about that and make a work around to use software rendering if nvidia GPU is used

ill look into it

**Comment by idea90** [comment]
adds for arch linux

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**